### PR TITLE
Clean up TimedMemberState - remove all metric related fields

### PR DESCRIPTION
**Pull request by puzpuzpuz**
* Implements https://github.com/hazelcast/management-center/issues/3334 by removing all metrics (probes) related fields from JSON representation of `TimedMemberState`
* Keeps the overall structure of JSON representation of TMS compatible with IMDG 4.0, as MC 4.1 will have to be compatible with both IMDG 4.0 and 4.1
* Also removes `WanSyncState` from `MemberState` as it was replaced with WAN events and not consumed anymore on MC side

MC counterpart PR: https://github.com/hazelcast/management-center/pull/3302
Potential next step: https://github.com/hazelcast/management-center/issues/3339

### CODE REVIEW

**Comment by hz-highfive** [comment]
The job `Hazelcast-pr-builder` of your PR failed ([log](
https://j-artifacts.s3.amazonaws.com/Hazelcast-pr-builder/1057/build_log.txt), [artifacts](https://j-artifacts.s3.amazonaws.com/Hazelcast-pr-builder/1057/Hazelcast-pr-builder-1057.zip)). 
Through arcane magic we have determined that the following fragments from the build log may contain information about the problem.
<details>
<summary>Click to expand the log file</summary>
<pre>
--------------------------
-------TEST FAILURE-------
--------------------------
[INFO] Results:
[INFO] 
[ERROR] Failures: 
[ERROR]   ClientCacheEventJournalBasicTest>AbstractEventJournalBasicTest.allowReadingWithFutureSeq:512->HazelcastTestSupport.assertOpenEventually:1108->HazelcastTestSupport.assertOpenEventually:1123 CountDownLatch failed to complete within 30 seconds, count left: 1
[INFO] 
[ERROR] Tests run: 27761, Failures: 1, Errors: 0, Skipped: 1001
[INFO] 

[ERROR] There are test failures.
</details>
</pre>

**Comment by hz-highfive** [comment]
The job `Hazelcast-pr-builder` of your PR failed ([log](
https://j-artifacts.s3.amazonaws.com/Hazelcast-pr-builder/1058/build_log.txt), [artifacts](https://j-artifacts.s3.amazonaws.com/Hazelcast-pr-builder/1058/Hazelcast-pr-builder-1058.zip)). 
Through arcane magic we have determined that the following fragments from the build log may contain information about the problem.
<details>
<summary>Click to expand the log file</summary>
<pre>
--------------------------
-------TEST FAILURE-------
--------------------------
[INFO] Results:
[INFO] 
[ERROR] Failures: 
[ERROR]   ClientCacheEventJournalBasicTest>AbstractEventJournalBasicTest.allowReadingWithFutureSeq:512->HazelcastTestSupport.assertOpenEventually:1108->HazelcastTestSupport.assertOpenEventually:1123 CountDownLatch failed to complete within 30 seconds, count left: 1
[ERROR]   ClientMapEventJournalBasicTest>AbstractEventJournalBasicTest.allowReadingWithFutureSeq:512->HazelcastTestSupport.assertOpenEventually:1108->HazelcastTestSupport.assertOpenEventually:1123 CountDownLatch failed to complete within 30 seconds, count left: 1
[ERROR]   LocalWanPublisherStatsTest.testSerialization:49 expected:<1> but was:<0>
[ERROR]   LocalWanStatsImplTest.testSerialization:69 expected:<1> but was:<0>
[INFO] 
[ERROR] Tests run: 27761, Failures: 4, Errors: 0, Skipped: 1001
[INFO] 

[ERROR] There are test failures.
</details>
</pre>

**Comment by hz-highfive** [comment]
The job `Hazelcast-pr-builder` of your PR failed ([log](
https://j-artifacts.s3.amazonaws.com/Hazelcast-pr-builder/1061/build_log.txt), [artifacts](https://j-artifacts.s3.amazonaws.com/Hazelcast-pr-builder/1061/Hazelcast-pr-builder-1061.zip)). 
Through arcane magic we have determined that the following fragments from the build log may contain information about the problem.
<details>
<summary>Click to expand the log file</summary>
<pre>
--------------------------
-------TEST FAILURE-------
--------------------------
[INFO] Results:
[INFO] 
[ERROR] Failures: 
[ERROR]   ClientCacheEventJournalBasicTest>AbstractEventJournalBasicTest.allowReadingWithFutureSeq:512->HazelcastTestSupport.assertOpenEventually:1108->HazelcastTestSupport.assertOpenEventually:1123 CountDownLatch failed to complete within 30 seconds, count left: 1
[INFO] 
[ERROR] Tests run: 27761, Failures: 1, Errors: 0, Skipped: 1001
[INFO] 

[ERROR] There are test failures.
</details>
</pre>

File: hazelcast/src/main/java/com/hazelcast/internal/monitor/impl/MemberPartitionStateImpl.java [hunk 1]
```diff
@@ -37,6 +37,7 @@
 
     List<Integer> partitions = new ArrayList<Integer>(DEFAULT_PARTITION_COUNT);
     boolean memberStateSafe;
+    @Deprecated
```
Comment (emre-aydin) [inline hazelcast/src/main/java/com/hazelcast/internal/monitor/impl/MemberPartitionStateImpl.java]: Why is it deprecated instead of removed?

Comment (emre-aydin) [reply hazelcast/src/main/java/com/hazelcast/internal/monitor/impl/MemberPartitionStateImpl.java]: I see that it is removed now.

Comment (puzpuzpuz) [reply hazelcast/src/main/java/com/hazelcast/internal/monitor/impl/MemberPartitionStateImpl.java]: It was an overlook. I've just pushed be0c0bd1e6d7d4a207c55d92793ca8f8a0a6e601 which fixes it.

**Comment by hz-highfive** [comment]
The job `Hazelcast-pr-builder` of your PR failed ([log](
https://j-artifacts.s3.amazonaws.com/Hazelcast-pr-builder/1065/build_log.txt), [artifacts](https://j-artifacts.s3.amazonaws.com/Hazelcast-pr-builder/1065/Hazelcast-pr-builder-1065.zip)). 
Through arcane magic we have determined that the following fragments from the build log may contain information about the problem.
<details>
<summary>Click to expand the log file</summary>
<pre>
--------------------------
-------TEST FAILURE-------
--------------------------
[INFO] Results:
[INFO] 
[ERROR] Failures: 
[ERROR]   ClientCacheEventJournalBasicTest>AbstractEventJournalBasicTest.allowReadingWithFutureSeq:512->HazelcastTestSupport.assertOpenEventually:1108->HazelcastTestSupport.assertOpenEventually:1123 CountDownLatch failed to complete within 30 seconds, count left: 1
[ERROR]   ClientMapEventJournalBasicTest>AbstractEventJournalBasicTest.allowReadingWithFutureSeq:512->HazelcastTestSupport.assertOpenEventually:1108->HazelcastTestSupport.assertOpenEventually:1123 CountDownLatch failed to complete within 30 seconds, count left: 1
[INFO] 
[ERROR] Tests run: 27748, Failures: 2, Errors: 0, Skipped: 1001
[INFO] 

[ERROR] There are test failures.
</details>
</pre>

**Comment by hz-highfive** [comment]
The job `Hazelcast-pr-builder` of your PR failed ([log](
https://j-artifacts.s3.amazonaws.com/Hazelcast-pr-builder/1066/build_log.txt), [artifacts](https://j-artifacts.s3.amazonaws.com/Hazelcast-pr-builder/1066/Hazelcast-pr-builder-1066.zip)). 
Through arcane magic we have determined that the following fragments from the build log may contain information about the problem.
<details>
<summary>Click to expand the log file</summary>
<pre>
--------------------------
-------TEST FAILURE-------
--------------------------
[INFO] Results:
[INFO] 
[ERROR] Failures: 
[ERROR]   ClientCacheEventJournalBasicTest>AbstractEventJournalBasicTest.allowReadingWithFutureSeq:512->HazelcastTestSupport.assertOpenEventually:1108->HazelcastTestSupport.assertOpenEventually:1123 CountDownLatch failed to complete within 30 seconds, count left: 1
[ERROR]   ClientReliableTopicOnClusterRestartTest.shouldFail_OnClusterRestart_whenDataLoss_notLossTolerant:185->lambda$shouldFail_OnClusterRestart_whenDataLoss_notLossTolerant$5:187
[INFO] 
[ERROR] Tests run: 27748, Failures: 2, Errors: 0, Skipped: 1001
[INFO] 

[ERROR] There are test failures.
</details>
</pre>

File: hazelcast/src/main/java/com/hazelcast/internal/management/TimedMemberStateFactory.java [hunk 2]
```diff
@@ -244,195 +222,153 @@ protected void createNodeState(MemberStateImpl memberState) {
...
         }
-        return count;
+        memberState.setTopicsWithStats(topicsWithStats);
     }
 
-    private int handleQueue(MemberStateImpl memberState, int count, Config config, Map<String, LocalQueueStats> queues) {
-        for (Map.Entry<String, LocalQueueStats> entry : queues.entrySet()) {
-            String name = entry.getKey();
+    private void handleQueue(MemberStateImpl memberState, Config config, Map<String, LocalQueueStats> queues) {
+        Set<String> mapsWithStats = createHashSet(queues.size());
```
Comment (blazember) [inline hazelcast/src/main/java/com/hazelcast/internal/management/TimedMemberStateFactory.java:336]: Minor: maps -> queue

File: hazelcast/src/main/java/com/hazelcast/internal/management/TimedMemberStateFactory.java [hunk 3]
```diff
@@ -244,195 +222,153 @@ protected void createNodeState(MemberStateImpl memberState) {
...
         }
-        return count;
+        memberState.setQueuesWithStats(mapsWithStats);
     }
 
-    private int handleMap(MemberStateImpl memberState, int count, Config config, Map<String, LocalMapStats> maps) {
-        for (Map.Entry<String, LocalMapStats> entry : maps.entrySet()) {
-            String name = entry.getKey();
+    private void handleMap(MemberStateImpl memberState, Config config, Map<String, LocalMapStats> maps) {
+        Set<String> queuesWithStats = createHashSet(maps.size());
```
Comment (blazember) [inline hazelcast/src/main/java/com/hazelcast/internal/management/TimedMemberStateFactory.java:346]: Minor: queues -> maps

File: hazelcast/src/main/java/com/hazelcast/internal/monitor/MemberState.java [hunk 4]
```diff
@@ -58,34 +51,10 @@
...
-
-    LocalPNCounterStats getLocalPNCounterStats(String pnCounterName);
-
-    LocalReplicatedMapStats getLocalReplicatedMapStats(String replicatedMapName);
-
-    LocalExecutorStats getLocalExecutorStats(String executorName);
-
-    LocalCacheStats getLocalCacheStats(String cacheName);
-
     LocalWanStats getLocalWanStats(String schemeName);
```
Comment (blazember) [inline hazelcast/src/main/java/com/hazelcast/internal/monitor/MemberState.java:54]: It seems this method is not called.

File: hazelcast/src/main/java/com/hazelcast/internal/management/TimedMemberStateFactory.java (hunk 2 above)
Comment (puzpuzpuz) [reply hazelcast/src/main/java/com/hazelcast/internal/management/TimedMemberStateFactory.java]: Arhhhh, I confused `handleMap()` with this method when renaming local variables. 🙃

Addressed in 3bae9fb45929a28d57d909a33aa7192c11018b94

File: hazelcast/src/main/java/com/hazelcast/internal/management/TimedMemberStateFactory.java (hunk 3 above)
Comment (puzpuzpuz) [reply hazelcast/src/main/java/com/hazelcast/internal/management/TimedMemberStateFactory.java]: Addressed in 3bae9fb45929a28d57d909a33aa7192c11018b94

File: hazelcast/src/main/java/com/hazelcast/internal/monitor/MemberState.java (hunk 4 above)
Comment (puzpuzpuz) [reply hazelcast/src/main/java/com/hazelcast/internal/monitor/MemberState.java]: Yup, and it was like this before the PR if I'm not mistaken. This whole interface feels artificial.

Removed the method in 3bae9fb45929a28d57d909a33aa7192c11018b94

**Comment by puzpuzpuz** [comment]
@blazember could you take another look?

**Comment by hz-highfive** [comment]
The job `Hazelcast-pr-builder` of your PR failed ([log](
https://j-artifacts.s3.amazonaws.com/Hazelcast-pr-builder/1074/build_log.txt), [artifacts](https://j-artifacts.s3.amazonaws.com/Hazelcast-pr-builder/1074/Hazelcast-pr-builder-1074.zip)). 
Through arcane magic we have determined that the following fragments from the build log may contain information about the problem.
<details>
<summary>Click to expand the log file</summary>
<pre>
--------------------------
-------TEST FAILURE-------
--------------------------
[INFO] Results:
[INFO] 
[ERROR] Failures: 
[ERROR]   ClientCacheEventJournalBasicTest>AbstractEventJournalBasicTest.allowReadingWithFutureSeq:512->HazelcastTestSupport.assertOpenEventually:1108->HazelcastTestSupport.assertOpenEventually:1123 CountDownLatch failed to complete within 30 seconds, count left: 1
[ERROR]   ClientMapEventJournalBasicTest>AbstractEventJournalBasicTest.allowReadingWithFutureSeq:512->HazelcastTestSupport.assertOpenEventually:1108->HazelcastTestSupport.assertOpenEventually:1123 CountDownLatch failed to complete within 30 seconds, count left: 1
[ERROR]   ClientReliableTopicOnClusterRestartTest.shouldFail_OnClusterRestart_whenDataLoss_notLossTolerant:185->lambda$shouldFail_OnClusterRestart_whenDataLoss_notLossTolerant$5:187
[INFO] 
[ERROR] Tests run: 27748, Failures: 3, Errors: 0, Skipped: 1001
[INFO] 

[ERROR] There are test failures.
</details>
</pre>

**Comment by puzpuzpuz** [comment]
Failed tests are known to be flaky: #16964, #17015. Thus, I'm merging this PR.

**Pull request merged**